integration-libs/epd-visualization/schematics/add-epd-visualization/index.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  chain,
+  noop,
+  Rule,
+  SchematicContext,
+  Tree,
+} from '@angular-devkit/schematics';
+import {
+  addLibraryFeature,
+  addPackageJsonDependenciesForLibrary,
+  CLI_EPD_VISUALIZATION_FEATURE,
+  CustomConfig,
+  EPD_VISUALIZATION_CONFIG,
+  readPackageJson,
+  shouldAddFeature,
+  SPARTACUS_EPD_VISUALIZATION,
+  validateSpartacusInstallation,
+} from '@spartacus/schematics';
+import { parse, stringify } from 'comment-json';
+import { peerDependencies } from '../../package.json';
+import {
+  EPD_VISUALIZATION_FEATURE_NAME_CONSTANT,
+  EPD_VISUALIZATION_FOLDER_NAME,
+  EPD_VISUALIZATION_MODULE,
+  EPD_VISUALIZATION_MODULE_NAME,
+  EPD_VISUALIZATION_ROOT_MODULE,
+  EPD_VISUALIZATION_TRANSLATIONS,
+  EPD_VISUALIZATION_TRANSLATION_CHUNKS_CONFIG,
+  SCSS_FILE_NAME,
+  SPARTACUS_EPD_VISUALIZATION_ROOT,
+  SPARTACUS_EPD_VISUALIZATION_ASSETS,
+} from '../constants';
+import { Schema as SpartacusEpdVisualizationOptions } from './schema';
+
+const TSConfigFileName = 'tsconfig.json';
+
+export function addEpdVisualizationFeature(
+  options: SpartacusEpdVisualizationOptions
+): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    const packageJson = readPackageJson(tree);
+    validateSpartacusInstallation(packageJson);
+
+    return chain([
+      addPackageJsonDependenciesForLibrary(peerDependencies, options),
+
+      shouldAddFeature(CLI_EPD_VISUALIZATION_FEATURE, options.features)
+        ? chain([addEpdVisualization(options), updateTsConfig()])
+        : noop(),
+    ]);
+  };
+}
+
+function addEpdVisualization(options: SpartacusEpdVisualizationOptions): Rule {
+  const customConfig: CustomConfig[] = [
+    {
+      import: [
+        {
+          moduleSpecifier: SPARTACUS_EPD_VISUALIZATION_ROOT,
+          namedImports: [EPD_VISUALIZATION_CONFIG],
+        },
+      ],
+      content: `<${EPD_VISUALIZATION_CONFIG}>{
+        epdVisualization: {
+          ui5: {
+            bootstrapUrl: "https://sapui5.hana.ondemand.com/1.97.0/resources/sap-ui-core.js"
```

Comment:
I can see customers can choose their own URL (and version) of UI5. Does our feature implementation depend on UI5 being specifically in version 1.97? If yes, we should document it in the official docs of the feature.

Reply:
There is an issue affecting hotspot selection colour that is fixed in 1.97.0 but not in any previous release of UI5.
I have tested with snapshot versions of UI5 1.98.0 (which has not been released yet).

We will want to update to UI5 1.98.0 or later in the next Spartacus major release, since 1.98.0 contains a number of fixes relating to animation playback.

We can't stop a customer from changing this value, but we can make a statement regarding versions that are supported.

I think the approach should be that a particular version is set at the time of a particular Spartacus release (which would be the version supported with that particular Spartacus release).

Reply:
This will be problematic if in future Spartacus releases you rely on SapUi being upgraded to 1.xx or 2.yy, but in customers' app's code contains still the initially set, old url 1.97 in their app's configuraitons. 

What do you think about letting customers configure only the base URL of ui5, and let configure the version separately?
Then we would build the full URL to sap-ui.core.js from parts.

We could provide the default version ourselves. Thanks to that we could control it and update in the next Spartacus releases ourselves, without interventing in customer's app code (via another cumbersome update-schematics).

example in our lib:
```ts
provideDefaultConfig({
  epdVisualisation:
    ui5: { 
      baseUrl: 'https://sapui5.hana.ondemand.com/',
      version: '1.97.0'
    }
 })
 ```
 
 What do you think about it?

Reply:
... But if you think it's a common case for customers to provide their custom full URL to UI5 (instead of `sapui5....demand.com` , then it would be an argument against splitting the URL config into parts.